code/modules/mob/living/simple_animal/hostile/swarmer.dm

```diff
@@ -0,0 +1,834 @@
+  */
+/mob/living/simple_animal/hostile/swarmer/proc/CreateTrap()
+	set name = "Create trap"
+	set category = "Swarmer"
+	set desc = "Creates a simple trap that will non-lethally electrocute anything that steps on it. Costs 4 resources."
+	if(locate(/obj/structure/swarmer/trap) in loc)
+		to_chat(src, "<span class='warning'>There is already a trap here. Aborting.</span>")
+		return
+	if(resources < 4)
+		to_chat(src, "<span class='warning'>We do not have the resources for this!</span>")
+		return
+	Fabricate(/obj/structure/swarmer/trap, 4)
+
+/**
+  * Called when a swarmer attempts to create a barricade
+  *
+  * Proc used to allow a swarmer to create a barricade.  Checks if a barricade is on the tile, then if the swarmer can afford it, and then will attempt to create a barricade after a second delay.
+  */
+/mob/living/simple_animal/hostile/swarmer/proc/CreateBarricade()
+	set name = "Create barricade"
+	set category = "Swarmer"
+	set desc = "Creates a barricade that will stop anything but swarmers and disabler beams from passing through.  Costs 4 resources."
+	if(locate(/obj/structure/swarmer/blockade) in loc)
+		to_chat(src, "<span class='warning'>There is already a blockade here. Aborting.</span>")
+		return
+	if(resources < 4)
+		to_chat(src, "<span class='warning'>We do not have the resources for this!</span>")
+		return
+	if(do_mob(src, src, 10))
+		Fabricate(/obj/structure/swarmer/blockade, 4)
+
+
+/obj/structure/swarmer/blockade
+	name = "swarmer blockade"
+	desc = "A quickly assembled energy blockade. Will not retain its form if damaged enough, but disabler beams and swarmers pass right through."
+	icon_state = "barricade"
+	light_range = MINIMUM_USEFUL_LIGHT_RANGE
+	max_integrity = 50
+
+/obj/structure/swarmer/blockade/CanAllowThrough(atom/movable/O)
+	. = ..()
+	if(isswarmer(O))
+		return TRUE
+	if(istype(O, /obj/projectile/beam/disabler))
+		return TRUE
+
+/**
+  * Called when a swarmer attempts to create a drone
+  *
+  * Proc used to allow a swarmer to create a drone.  Checks if the swarmer can afford the drone, then creates it after 5 seconds, and also registers it to the creating swarmer so it can command it
+  */
+/mob/living/simple_animal/hostile/swarmer/proc/CreateSwarmer()
+	set name = "Replicate"
+	set category = "Swarmer"
+	set desc = "Creates a duplicate of ourselves, capable of protecting us while we complete our objectives."
+	to_chat(src, "<span class='info'>We are attempting to replicate ourselves. We will need to stand still until the process is complete.</span>")
+	if(resources < 20)
+		to_chat(src, "<span class='warning'>We do not have the resources for this!</span>")
+		return
+	if(!isturf(loc))
+		to_chat(src, "<span class='warning'>This is not a suitable location for replicating ourselves. We need more room.</span>")
+		return
+	if(do_mob(src, src, 50))
+		var/createtype = SwarmerTypeToCreate()
+		if(createtype)
+			dronelist += Fabricate(createtype, 20)
```

Comment:
Register to their destruction not to keep a reference of a qdeleting or deleted object, also avoiding a hard delete.